pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -76,22 +94,22 @@ func (r *Client) CheckSnapshotReady(vmRef ref.Ref, snapshot string) (ready bool,
 		err = liberr.Wrap(err)
 		return
 	}
-	jobs, err := r.getJobs(correlationID)
+	events, err := r.getEvents(correlationID)
 	if err != nil {
 		err = liberr.Wrap(err)
 		return
 	}
-	if len(jobs) < 1 {
-		err = liberr.New("No jobs found for correlation ID", "correlationID", correlationID)
+	if len(events) < 1 {
+		err = liberr.New("No event found for correlation ID", "correlationID", correlationID)
 		return
 	}
 	ready = true
-	for _, job := range jobs {
-		switch job.MustStatus() {
-		case ovirtsdk.JOBSTATUS_FAILED, ovirtsdk.JOBSTATUS_ABORTED:
+	for _, event := range events {
+		switch event.MustCode() {
+		case SNAPSHOT_FINISHED_FAILURE:
 			err = liberr.New("Snapshot creation failed!", "correlationID", correlationID)
 			ready = false
-		case ovirtsdk.JOBSTATUS_STARTED, ovirtsdk.JOBSTATUS_UNKNOWN:
+		case CREATE_SNAPSHOT:
```

Comment:
is this logic correct - with jobs, a job has a single status - meaning that a job with status `JOBSTATUS_STARTED` means there's an ongoing operation that didn't succeed, but here we can get an event of `CREATE_SNAPSHOT` followed by `SNAPSHOT_FINISHED_SUCCESS`, which means the operation succeeded and yet we'll set ready=false, no?

Reply:
Ahh good point! Added switch case for `SNAPSHOT_FINISHED_SUCCESS`. So if the `CREATE_SNAPSHOT` sets the ready to false and there are more events it will rewrite the ready to true

Reply:
are the events ordered?

Reply:
yes, from the go-ovirt client
```
// The following events occur:
// * id="1" - The API logs in the admin user account.
// * id="2" - The API logs out of the admin user account.
// The order of the returned list of events is always garanteed. If the `sortby` clause is included in the
// `search` parameter, then the events will be ordered according to that clause.
...
```